solc/CommandLineInterface.cpp

```diff
@@ -1085,23 +1101,28 @@ bool CommandLineInterface::processInput()
 
 	if (m_args.count(g_argAssemble) || m_args.count(g_argStrictAssembly) || m_args.count(g_argYul))
 	{
+		vector<string> const nonAssemblyModeOptions = {
```

Comment:
Aren't there many more?

Reply:
There are but I did not want to dig that deeply into this code. I mostly wanted to deal with `--output-dir` in assembly mode (I originally thought it was broken) and along the way managed to figure out how which other options are related to assembly mode and how so I didn't want that knowledge to go to waste. I know for sure that the ones I added were being ignored but I'm a bit fuzzy about some others. I could make a more comprehensive PR if you think it's worthwhile but I'd need to spend a bit more time on it.

Reply:
Can you make a short comment in the code saying that this is not the full list?

Reply:
Good idea. Done.

Reply:
Would it make sense to also create an issue for it?
Or is the `TODO` comment enough?

Reply:
If we want to change is so that eventually invalid options are always rejected rather than ignored then it would certainly make sense. Right now it isn't consistent so I wasn't even sure if there wouldn't be some objections to this PR on that basis. After all the help text says things like "Switch to assembly mode, ignoring all options except ...".

@chriseth Is this the direction in which we want to go? I would be certainly for it.

Reply:
We might do that one time, but I think it is not too important right now.

Reply:
In that case I think that the TODO is enough.